associate.py

```diff
@@ -0,0 +1,67 @@
+import argparse
+import os
+
+
+def read_file_list(filename, isRGB):
+    file = open(filename)
+    data = file.read()
+    lines = data.replace(","," ").replace("\t"," ").split("\n")
+    lines = [line for line in lines if line != '']
+    lines = list(map(float, lines))
+    files = None
+    if isRGB:
+        files = os.listdir(os.path.join(os.path.dirname(filename), "color"))
+    else:
+        files = os.listdir(os.path.join(os.path.dirname(filename), "depth"))
+    if isRGB:
+        files = [os.path.join("color", x) for x in files]
+    else:
+        files = [os.path.join("depth", x) for x in files]
+    files.sort()
+    d = dict(zip(lines, files))
+    return d
+
+
+def associate(first_list, second_list,offset,max_difference):
+    first_keys = list(first_list.keys())
+    second_keys = list(second_list.keys())
+    potential_matches = [(abs(a - (b + offset)), a, b) 
+                         for a in first_keys 
```

Comment:
Double loops are always scary, use vector operations as much as possible.
Pseudo code for finding match for one key from `first_keys`.

```
second_keys = np.asarray(second_keys)
ts = first_keys[0]
best_ind = np.argmin(np.abs(second_keys + offset - ts))
```